cannot generate tsx types

### Problem

`dts` does not correctly generate types for `tsx` files when using a specific `tsconfig`.


```sh
# File Struct
.
├── src
├── tsconfig.example.json
├── tsconfig.json
├── tsconfig.src.json
└── vite.config.ts

```


```jsonc
// tsconfig.json
{
  "files": [],
  "references": [
    { "path": "./tsconfig.example.json" },
    { "path": "./tsconfig.src.json" }
  ]
}
```

```jsonc
// tsconfig.src.json
{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "ESNext",
    "skipLibCheck": true,

    "moduleResolution": "bundler",
    "allowImportingTsExtensions": true,
    "noUncheckedIndexedAccess": true,
    "isolatedModules": true,
    "noImplicitOverride": true,
    "moduleDetection": "force",
    "noEmit": true,
    "jsx": "react-jsx",
    ^^^^^^^^^^^^^^^^^^^
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["src"]
}

```

## run command

```sh
pkgroll -p tsconfig.src.json

# got an error
src/index.ts(2,15): error TS6142: Module './components/stats' was resolved to 
'/xxx/src/components/stats/index.tsx', but '--jsx' is not set.
                                       ^^^^^^^^^^^^^^^^^^^^^^^
```


https://github.com/privatenumber/pkgroll/blob/2a988e454bd8fd163b567e5fa2e67ab89864c3f3/src/utils/get-rollup-configs.ts#L61


### Expected behavior

I found that it was because the specified tsconfig path was not passed to `rollup-plugin-dts`, so `rollup-plugin-dts` used the root path's `tsconfig.json` by default

Just pass the specified `tsconfig` path to `rollup-plugin-dts` to fix this problem.




### Minimal reproduction URL

https://stackblitz.com/edit/stackblitz-starters-ayudec

### Version

v2.5.0

### Node.js version

v22.6.0

### Package manager

pnpm

### Operating system

macOS

### Bugs are expected to be fixed by those affected by it

- [X] I'm interested in working on this issue

### Compensating engineering work financially will speed up resolution

- [ ] I'm willing to offer financial support

--- a/src/utils/get-rollup-configs.ts
+++ b/src/utils/get-rollup-configs.ts
@@ -80,6 +80,7 @@ const getConfig = {
 						module: ts.default.ModuleKind.Preserve,
 						moduleResolution: ts.default.ModuleResolutionKind.Bundler,
 					},
+					tsconfig: tsconfig?.path,
 				}) as Plugin,
 			],
 			output: [] as unknown as Output,

--- a/tests/specs/builds/output-types.ts
+++ b/tests/specs/builds/output-types.ts
@@ -450,5 +450,37 @@ export default testSuite(({ describe }, nodePath: string) => {
 			expect(types).toMatch('\'dep-a\'');
 			expect(types).toMatch('.data');
 		});
+
+		test('custom tsconfig.json path', async () => {
+			await using fixture = await createFixture({
+				...packageFixture({
+					installTypeScript: true,
+					installReact: true,
+				}),
+				'package.json': createPackageJson({
+					types: './dist/component.d.ts',
+					peerDependencies: {
+						react: '*',
+					},
+				}),
+				'tsconfig.custom.json': createTsconfigJson({
+					compilerOptions: {
+						jsx: 'react-jsx',
+					},
+				}),
+			});
+
+			const pkgrollProcess = await pkgroll(['-p', 'tsconfig.custom.json'], {
+				cwd: fixture.path,
+				nodePath,
+			});
+
+			expect(pkgrollProcess.exitCode).toBe(0);
+			expect(pkgrollProcess.stderr).toBe('');
+
+			const content = await fixture.readFile('dist/component.d.ts', 'utf8');
+			expect(content).toMatch('declare const Component: () => react_jsx_runtime.JSX.Element');
+			expect(content).toMatch('export { Component }');
+		});
 	});
 });

